### PR TITLE
RUMM-2969 SR Add new capability to Recorder for EditTextView and CheckedEditTextView imp

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

Currently all the `EditTextView` and `CheckedEditTextView` implementations were resolved as a pure `TextView` without the extra visual elements (checkbox or text underline). In this PR we are adding a new capability to the SR `View -To -Wireframe` mappers to resolve these components as a collection of wireframes:

1. An implementation of an `EditTextView` class will be resolved as a `TextWireframe` + `ShapeWireframe` for the underline
2. An implementation of a `CheckedEditTextView` will be resolved as a `TextWireframe` + `ShapeWireframe` for the checkbox mark

Please have in mind that there will be probably several new iterations over this current code once we detect potential corner cases. This part only cover the basics.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/ViewUtils.kt [hunk 1]
```diff
@@ -43,4 +43,16 @@ internal class ViewUtils {
                         .isAssignableFrom(view::class.java)
                 )
     }
+
```
Comment (mariusc83) [inline library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/ViewUtils.kt:46]: @reviewers : The only reason I can't make this class a Singleton is that it uses a state based on the `Build.Version` value when generating the lazy sets for `isSystemView` ids. This breaks my unit tests as once this class is instantiated those `sets` are generated only for the current state of the `Build.Version` so I cannot re - initialize it. I could move those sets inside the `ViewUtils#checkIfSystemNoise` methods but this is called for each view in the tree every time we take a snapshot so there will be too many sets generated. If you have a better idea let me know, if not I don't mind keeping it like that for now.

**Review by 0xnm — APPROVED** [review]
great! I left few suggestions, but they are not blocking for me.

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableWireframeMapper.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    internal fun resolveCheckableWireframe(view: T, pixelDensity: Float):
+        MobileSegment.Wireframe? {
+        val checkboxId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(
+            view,
+            CHECKBOX_KEY_NAME
+        )
+        return if (checkboxId != null) {
+            val viewGlobalBounds = resolveViewGlobalBounds(view, pixelDensity)
+            val checkBoxPosX = viewGlobalBounds.x + viewGlobalBounds.width
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableWireframeMapper.kt:28]: yeah that's a good question but I wanted to cover only the basics for now. I am pretty sure I will have to fix things here as I don't really know all the corner cases. For now I saw that the default way if on the right. Regarding the `height` I just considered that the checkbox.y should be the text.y and it looks fine in the player. Likewise I will probably have to add corrections here but I am waiting for visual feedback first.

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableWireframeMapper.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                )
+            } else {
+                null
+            }
+            MobileSegment.Wireframe.ShapeWireframe(
+                id = checkboxId,
+                x = checkBoxPosX,
+                y = viewGlobalBounds.y,
+                width = viewGlobalBounds.height,
+                height = viewGlobalBounds.height,
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableWireframeMapper.kt:43]: yeah...the default is square and usually the height is lower than the width. Like above I want to keep it simple at first and iterate over later as I really don't know exactly the best formula here.

**Pull request merged**